tee-worker/core-primitives/stf-executor/src/executor.rs

```diff
@@ -124,16 +126,20 @@ where
 
 		debug!("execute on STF, call with nonce {}", trusted_call.nonce);
 		let mut extrinsic_call_backs: Vec<OpaqueCall> = Vec::new();
-		if let Err(e) = Stf::execute_call(
+		let rpc_response_value = match Stf::execute_call(
 			state,
 			shard,
 			trusted_call.clone(),
+			trusted_operation.hash(),
 			&mut extrinsic_call_backs,
 			self.node_metadata_repo.clone(),
 		) {
-			error!("Stf execute failed: {:?}", e);
-			return Ok(ExecutedOperation::failed(top_or_hash))
-		}
+			Err(e) => {
+				error!("Stf execute failed: {:?}", e);
+				return Ok(ExecutedOperation::failed(top_or_hash, extrinsic_call_backs))
+			},
+			Ok(res) => res,
```

Comment:
Here we should only care about res coming from indirect calls

```
			Ok(res) => match trusted_operation {
				TrustedOperation::indirect_call(_) => Some(res),
				_ => None
			},
```

Reply:
same above - how about another PR dedicated for this problem?